x-pack/plugins/security/public/session/session_lifespan_warning.tsx

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiProgress } from '@elastic/eui';
+
+export const SessionLifespanWarning = () => {
+  return (
+    <>
+      <EuiProgress size="xs" color="danger" position="absolute" />
+      <p>
+        <FormattedMessage
+          id="xpack.security.components.sessionLifespanWarning.message"
+          defaultMessage="Your session will soon reach the maximum time limit. You will need to log in again."
+        />
```

Comment:
@legrego 

Can the message say when the session will expire?

Your session will expire in xxx seconds
Please [log in](url) to start a new session.

Reply:
@gchaps before I wrote this message I looked at some of the commit history; at one point the session idle timeout warning did display the time remaining. I'm not sure exactly when (I'd have to dig into the commit history), but it appears that was removed at a later point, so I opted not to add it back in.

@legrego do you have any thoughts on the matter / insight as to what the thinking was behind that decision?

Reply:
Do you have a link handy to the old countdown style message? That might jog my memory, otherwise I'm really not sure about its history.

I'm not opposed to a countdown timer FWIW

Reply:
I saw it in a screenshot in #19600, I assumed it was removed on purpose but it sounds like toasts simply don't provide the countdown out of the box.

I'll add a countdown in.